chunky/src/java/se/llbit/chunky/renderer/scene/Fog.java

```diff
@@ -48,6 +54,47 @@ public boolean fastFog() {
     return fastFog;
   }
 
+  public FogLayer[] getFogLayers() {
+    return layers;
+  }
+
+  public void addLayer() {
+    int length = layers.length;
+    layers = Arrays.copyOf(layers, layers.length + 1);
+    layers[length] = new FogLayer(scene);
+    scene.refresh();
+  }
+
+  public void removeLayer(int index) {
+    FogLayer[] layers2 = new FogLayer[layers.length - 1];
+    int currentIndex = 0;
+    int newIndex = 0;
+    for (FogLayer layer : layers) {
+      if (!(currentIndex == index)) {
+        layers2[newIndex] = layer;
+        newIndex++;
+      }
+      currentIndex++;
+    }
+    layers = Arrays.copyOf(layers2, layers2.length);
```

Comment:
Wouldn't it be way easier to use a list?

Reply:
I'll attempt the conversion. At some point, I did store it as an ArrayList, which was much easier, but I was tricked into thinking that it was causing an issue, where the issue was actually due to the minimum value of *0.01* for fog breadth, which makes fog rendering very unstable and likely to crash or hang forever. By then, I had already changed it back to Array, though.